src/shell/kill.c

```diff
@@ -42,21 +42,11 @@ static void kill_cb (flux_t *h, flux_msg_handler_t *mh,
                      const flux_msg_t *msg, void *arg)
 {
     flux_shell_t *shell = arg;
-    uint32_t userid;
-    uint32_t rolemask;
     int signum;
-
-    if (flux_msg_get_userid (msg, &userid) < 0
-        || flux_msg_get_rolemask (msg, &rolemask) < 0) {
-        log_err ("kill: unable to read msg userid or rolemask");
-        return;
-    }
-    if (!(rolemask & FLUX_ROLE_OWNER) && userid != getuid ()) {
-        log_msg ("kill: invalid userid %u", userid);
-        return;
-    }
     if (flux_msg_unpack (msg, "{s:i}", "signum", &signum) < 0) {
```

Comment:
Suggestion: `flux_event_unpack()`.

Reply:
Good suggestion. However, isn't `flux_msg_unpack() == flux_event_unpack()` if you don't pass storage for `topic`? (Here we don't need the topic, just the payload) And we already are guaranteed `FLUX_MSGTYPE_EVENT` because of `FLUX_MATCH_EVENT` used on msghandler.

Reply:
True, just for clarity I suppose, or the general argument that the event format could change.  Meh, not a big deal.

Reply:
Hm, good point. If event format might change, though, we might want to have `flux_event_t` instead of `flux_msg_t` to avoid confusion down the road..

Reply:
Not today, but point taken.